### ISSUE TITLE
Rule Change: no-promise-executor-return should allow returning resolve/reject

### ISSUE THREAD

**Issue by jdforsythe** [eslint/eslint#16123]
### What rule do you want to change?

no-promise-executor-return

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
new Promise((resolve, reject) => {
    getSomething((err, data) => {
        if (err) {
            return reject(err);
        }

        return resolve(data);
    });
});
```

### What does the rule currently do for this code?

error

### What will the rule do after it's changed?

Add an option to allow (or better yet, to _require_) `return` of the `reject()`/`resolve()` functions.

The rule already allows `return;` for control flow. Using `return reject(err)` or `return resolve(data)` is also used for control flow and it also prevents bugs where you could accidentally execute code after calling reject/resolve, e.g.:

```js
new Promise((resolve, reject) => {
    getSomething((err, data) => {
        if (err) {
           reject(err);
        }

        console.log('This executes after reject() is called');

       resolve(data);
    });
});
```

We require explicit returns on the `resolve()`/`reject()` functions to prevent this accidental execution of code.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
What version of ESLint are you using? I’m not getting an error with your example. 

I wouldn’t expect to get an error because the return statements in your example are not for the executor, they’re for a callback in the executor, which is a completely different scope and exempt from this check.

**Comment by jdforsythe** [comment]
Indeed it was a version issue and not a rule issue. Thank you.
